docs/DeveloperGuide.md

```diff
@@ -1,34 +1,357 @@
+XML files provide inbuilt tags to differentiate and classify members of the document. 
+With the use of XML file, it simplifies the parsing process of the cheatsheet files into the cheatsheet class, and vice versa.
+
+![](Images/Image9.PNG)
+
+Image 7: XML file showing the content of a cheatsheet file
+
+#### 3.2.4. Cheat Sheet Management (Adhy)
+
+All cheat sheets, both pre-loaded and user-defined, are stored in one static class called CheatSheetList during runtime. 
+Upon receiving a valid input from the user, the execute() method from Command class will invoke the mutation of CheatSheetList, 
+and then DataFileWriter will use the information in CheatSheetList to save it to the memory in the form of a txt file.
+
+![](Images/Image10.PNG)
+
+Image 8: Class diagram of CheatSheetList class
+
+The cheatSheets ArrayList is of type private, and setter/getter methods must be invoked to access the list. 
+For these setter and getter methods, it accepts both accessing by index and accessing by name to allow the user to easily search and fetch the cheat sheet. 
+When /list or /find command is invoked, printCheatSheetNames will return a string consisting all relevant cheat sheets to be printed by Printer class.
+
+
+#### 3.2.5. Data Storage (Theo)
+
+This feature allows the application to read and update data in the form of text files. 
+Having an external source to store data will allow the application to be able to retrieve it when it is relaunched at another instance. 
+This prevents the user from having to repeatedly create new cheatsheets and update the application settings each time he opens up CheatLogs. 
+
+
+This feature is split into 3 functions, which are indicated in Table 1 below.
+
+
+
+| Function | Purpose of Function |
+|----------|--------------------------------|
+|Read Files|Allows CheatLogs to extract cheatsheets and user-defined settings from text files provided that they follow a specified format.|
+|Write Files|Allows CheatLogs to write cheatsheets and user-defined settings to the hard-disk in the form of text files.|
+|Delete Files|Allows CheatLogs to remove unnecessary files from the hard-disk.|
+Table 1: Summary of the functions of data storage
+
+## 4. Implementation
+
+This section describes some noteworthy details on how certain features are implemented.
+
+### 4.1. Parsing of Data to Construct Commands
+
+The command classes follow the command pattern. 
+During construction in Parser.parse() they get the information they need to execute from parsed data based on the user input. 
+Flags in the input are used to separate the different information. (e.g. `/add /n if /l python /k is nice`, has 3 flags, /n /l and /k and the String “python" is associated with the flag /l). 
+This information is stored in a HashMap where the descriptor (defined in ArgumentEnum) of the flag is the key and the information associated with it is the value.
+The command can execute at a random time later via commandObj.execute().12
```

Comment:
I like that this section is included in the DG to allow readers to understand how the user input works.